renku/core/management/repository.py

```diff
@@ -504,17 +504,11 @@ def update_graphs(self, activity: Union[ProcessRun, WorkflowRun], database: Data
 
         provenance_graph.add(activity_collection)
 
-        # TODO: remove this try except once we move to the database and have a migration.
-        # Currently needed for 'test_graph()'
-        try:
-            for activity in activity_collection.activities:
-                database.add(activity)
-                database.add(activity.association.plan)
-
-            database.commit()
-        except AssertionError:
-            pass
+        for activity in activity_collection.activities:
+            database.get("activities").add(activity)
+            database.get("plans").add(activity.association.plan)
 
+        database.commit()
```

Comment:
This reminds me, I never checked if we need the `database.commit()` here, since the command with `with_database(write=True)` would do it at the end of the command.